sdk/js/src/aptos/api/tokenBridge.ts

```diff
@@ -0,0 +1,121 @@
+import { AptosAccount } from "aptos";
+import { assertChain } from "../../utils";
+import { AptosClientWrapper } from "../client";
+import { AptosNetwork, CONTRACT_ADDRESSES } from "../constants";
+import { deriveWrappedAssetAddress } from "../utils";
+import { AptosBaseApi } from "./base";
+
+export class AptosTokenBridgeApi extends AptosBaseApi {
+  constructor(client: AptosClientWrapper, network: AptosNetwork) {
+    super(client);
+    this.address = CONTRACT_ADDRESSES[network].token_bridge;
+  }
+
+  // Complete transfer
+
+  completeTransfer = (
+    sender: AptosAccount,
+    tokenChain: number,
+    tokenAddress: string,
+    vaa: Uint8Array,
+    feeRecipient: string,
+  ) => {
+    assertChain(tokenChain);
+    const assetContract = deriveWrappedAssetAddress(this.address, tokenChain, tokenAddress);
+    const payload = {
+      function: `${this.address}::complete_transfer::submit_vaa`,
+      type_arguments: [`${assetContract}::coin::T`],
+      arguments: [vaa, feeRecipient],
+    };
+    return this.client.executeEntryFunction(sender, payload);
+  };
+
+  completeTransferWithPayload = (
+    sender: AptosAccount,
+    tokenChain: number,
+    tokenAddress: string,
+    vaa: Uint8Array,
+  ) => {
+    assertChain(tokenChain);
+    const assetContract = deriveWrappedAssetAddress(this.address, tokenChain, tokenAddress);
+    const payload = {
+      function: `${this.address}::complete_transfer_with_payload::submit_vaa`,
+      type_arguments: [`${assetContract}::coin::T`],
+      arguments: [vaa],
+    };
+    return this.client.executeEntryFunction(sender, payload);
+  };
+
+  // Deploy coin
+
+  // don't need `signer` and `&signer` in argument list because the Move VM will inject them
+  deployCoin = (sender: AptosAccount) => {
+    const payload = {
+      function: `${this.address}::deploy_coin::deploy_coin`,
+      type_arguments: [],
+      arguments: [],
+    };
+    return this.client.executeEntryFunction(sender, payload);
+  };
+
+  // Register chain
+
+  registerChain = (sender: AptosAccount, vaa: Uint8Array) => {
+    const payload = {
+      function: `${this.address}::register_chain::submit_vaa`,
+      type_arguments: [],
+      arguments: [vaa],
+    };
+    return this.client.executeEntryFunction(sender, payload);
+  };
+
+  // Transfer tokens
+
+  transferTokensWithSigner = (
+    sender: AptosAccount,
+    tokenChain: number,
+    tokenAddress: string,
+    amount: number | bigint,
+    recipientChain: number | bigint,
+    recipient: Uint8Array,
+    relayerFee: number | bigint,
+    wormholeFee: number | bigint,
+    nonce: number | bigint,
+  ) => {
+    assertChain(tokenChain);
+    const assetContract = deriveWrappedAssetAddress(this.address, tokenChain, tokenAddress);
```

Comment:
currently this only handles wrapped assets, but it will need to handle native assets too. For native assets (i.e. when `tokenChain == aptos`), the `tokenAddress` field should be a fully qualified `address::module::type` string